procurement_auto_create_group/models/stock_rule.py

```diff
@@ -13,3 +15,13 @@ class StockRule(models.Model):
     def _onchange_group_propagation_option(self):
         if self.group_propagation_option != "propagate":
             self.auto_create_group = False
+
+    @api.model
+    def _prepare_auto_procurement_group_data(self):
+        name = self.env["ir.sequence"].next_by_code("procurement.group") or False
+        if not name:
+            raise UserError(_("No sequence defined for procurement group."))
+        return {
+            "name": name,
+            "partner_id": self.partner_address_id.id,
```

Comment:
As this function is declared as `api.model` this line will not work, I think.

Reply:
Indeed. Fixed.